### ISSUE TITLE
[Linux/arm32/arm64] corefx System.Runtime.Tests failure

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#21223]
JitStress=2

https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm/job/jitstress/job/arm_cross_checked_ubuntu_corefx_jitstress2_tst/37/consoleFull

```
20:00:20 >>>>> /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests
20:00:20 
20:00:20 Running tests in /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests/netcoreapp-Linux-Release-arm
20:00:20 timeout --kill-after=30s 20m ./RunTests.sh /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/testhost/netcoreapp-Linux-Release-arm
20:00:20 
20:00:20 ----- start 03:48:26 =============== To repro directly: =====================================================
20:00:20 pushd .
20:00:20 #!/usr/bin/env bash export COMPlus_TieredCompilation=0 export COMPlus_JitStress=2
20:00:20 chmod +x /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/testhost/netcoreapp-Linux-Release-arm/dotnet
20:00:20 /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/testhost/netcoreapp-Linux-Release-arm/dotnet xunit.console.dll System.Runtime.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
20:00:20 popd
20:00:20 ===========================================================================================================
20:00:20 /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests/netcoreapp-Linux-Release-arm /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests/netcoreapp-Linux-Release-arm
20:00:20 xUnit.net Console Runner v2.4.1-pre.build.4059 (32-bit .NET Core 4.6.27126.0)
20:00:20 [37m  Discovering: System.Runtime.Tests (method display = ClassAndMethod, method display options = None)
20:00:20 [m[37m  Discovered:  System.Runtime.Tests (found 4754 of 4801 test cases)
20:00:20 [m[37m  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 4)
20:00:20 [m[33;1m    System.Tests.IntPtrTests.Size [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.Add [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.Subtract [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.ImplicitCast [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.GetHashCodeRespectAllBits [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.Ctor_VoidPointer_ToPointer [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.IntPtrTests.Ctor_Long [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m* ExceptionTests - reported call stack:
20:00:20    at System.Tests.ExceptionTests.ThrowAndRethrowOtherMethod(ValueTuple`3& rethrownExceptionStackFrame) in /mnt/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2/_/fx/src/System.Runtime/tests/System/ExceptionTests.cs:line 156
20:00:20    at System.Tests.ExceptionTests.ThrowStatementDoesNotResetExceptionStackLineOtherMethod() in /mnt/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2/_/fx/src/System.Runtime/tests/System/ExceptionTests.cs:line 142
20:00:20 
[31;1m    System.Tests.ExceptionTests.ThrowStatementDoesNotResetExceptionStackLineOtherMethod [FAIL]
20:00:20 [m[37m      Assert.Equal() Failure
20:00:20 [m[37m                     ↓ (pos 5)
20:00:20 [m[37m      Expected: ThrowAndRethrowOtherMethod
20:00:20 [m[37m      Actual:   ThrowStatementDoesNotResetExceptionStackLineOt···
20:00:20 [m[37m                     ↑ (pos 5)
20:00:20 [m[30;1m      Stack Trace:
20:00:20 [m[37m        /mnt/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2/_/fx/src/System.Runtime/tests/System/ExceptionTests.cs(188,0): at System.Tests.ExceptionTests.VerifyCallStack(ValueTuple`3 expectedStackFrame, String reportedCallStack, Int32 skipFrames)
20:00:20 [m[37m        /mnt/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2/_/fx/src/System.Runtime/tests/System/ExceptionTests.cs(144,0): at System.Tests.ExceptionTests.ThrowStatementDoesNotResetExceptionStackLineOtherMethod()
20:00:20 [m[33;1m    System.Tests.ArgIteratorTests.ArgIterator_GetNextArgType [SKIP]
20:00:20 [m[37m      Condition(s) not met: "IsArgIteratorSupported"
20:00:20 [m[33;1m    System.Tests.ArgIteratorTests.ArgIterator_GetRemainingCount_GetNextArg [SKIP]
20:00:20 [m[37m      Condition(s) not met: "IsArgIteratorSupported"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.GetHashCodeRespectAllBits [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.Subtract [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.TestCtor_VoidPointer_ToPointer [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.Add [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.TestSize [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.Ctor_ULong [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[33;1m    System.Tests.UIntPtrTests.TestImplicitCast [SKIP]
20:00:20 [m[37m      Condition(s) not met: "Is64Bit"
20:00:20 [m[37m  Finished:    System.Runtime.Tests
20:00:20 [m[37m=== TEST EXECUTION SUMMARY ===
20:00:20 [m[37m   System.Runtime.Tests  Total: 29524, Errors: 0, Failed: 1, Skipped: 16, Time: 660.761s
20:00:20 [m[m/ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests/netcoreapp-Linux-Release-arm
20:00:20 ----- end 04:00:19 ----- exit code 1 ----------------------------------------------------------
20:00:20 Looking around for any Linux dump...
20:00:20 ... found no dump in /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_corefx_jitstress2_tst/_/fx/artifacts/bin/tests/System.Runtime.Tests/netcoreapp-Linux-Release-arm
20:00:20 
error: One or more tests failed while running tests from ''.  Exit code 1.
20:00:20 <<<<<
```

**Comment by BruceForstall** [comment]
The ThrowStatementDoesNotResetExceptionStackLineOtherMethod failure also happens on other platforms, e.g., arm64:

https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm64/job/jitstress/job/arm64_cross_checked_ubuntu16.04_corefx_jitstress2_tst/52/consoleFull

and x64:

https://github.com/dotnet/coreclr/issues/20799

**Comment by erozenfeld** [comment]
dotnet/corefx#34653 fixed this issue. I verified the fix on x64. Will verify on arm and arm64 and then remove exclusions and close this bug.

**Comment by erozenfeld** [comment]
arm and arm64 sill have failures in System.Runtime.Tests, tracked by #21022 and #17584.

**Comment by erozenfeld** [comment]
Closing this issue since ThrowStatementDoesNotResetExceptionStackLineOtherMethod is fixed by dotnet/corefx#34653.
